`planck`: fix deps and build failure

planar opened the pull request:
There is a missing dependency on `type_conv` (through `sexplib`) and a patch for the OMakefile problem (reported upstream at https://bitbucket.org/camlspotter/planck/issue/2/planck-211-fails-to-build-in-opam).

yallop commented:
That's unusual.  The [OSX job](https://travis-ci.org/ocaml/opam-repository/jobs/55683833) appears to have run on Ubuntu:

```
Using worker: worker-linux-da6418d7-2.bb.travis-ci.org:travis-linux-7
...
$ gcc --version
gcc (Ubuntu/Linaro 4.6.3-1ubuntu5) 4.6.3
...
sudo: hdiutil: command not found
```

planar commented:
I'm having trouble with this patch: it works when I apply it by hand to the `planck` sources, but not when I add it to the `opam` file :-(

I'm investigating.

planar commented:
I updated the dependencies for both `planck.2.1.1` and `planck.2.1.0` and I switched from a patch to a few build instructions to make `2.1.1` work. Let's see what Travis says.
